receptorctl/tests/test_workunit.py

```diff
@@ -0,0 +1,100 @@
+        """Wait for a workunit to finish
+
+        Args:
+            node_controller: The node controller used to create the workunit
+            unitid: The unitid of the workunit to wait for
+            timeout_seconds: The number of seconds to wait before timing out
+
+        Returns:
+            True if the workunit finished, False if it timed out
+        """
+        for _ in range(timeout_seconds):
+            status = node_controller.simple_command("work status {}".format(unitid))
+            state = status["StateName"]
+            if state == "Succeeded":
+                return True
+            else:
+                time.sleep(1)
+        time.sleep(1)
+        return False
+
+    return _wait_for_work_finished
+
+
+@pytest.mark.usefixtures("receptor_mesh_mesh1")
+class TestWorkUnit:
+    def test_workunit_simple(
+        self,
+        invoke_as_json,
+        default_receptor_controller_socket_file,
+        wait_for_work_finished,
+    ):
+        # Spawn a long running command
+        node1_controller = default_receptor_controller_socket_file
+        status = node1_controller.simple_command("status")
+
+        wait_for = 5  # in seconds
+
+        payload = "That's a long string example! And there's emoji too! 👾"
+        work = node1_controller.submit_work("echo-uppercase", payload, node="node3")
+        state_result = work.pop("result")
+        state_unitid = work.pop("unitid")
+
+        assert state_result == "Job Started"
+        assert wait_for_work_finished(
+            node1_controller, state_unitid, wait_for
+        ), "Workunit timed out and never finished"
+
+        work_result = (
+            node1_controller.get_work_results(state_unitid)
+            .read()
+            .decode("utf-8")
+            .strip()
+        )
+
+        assert (
+            payload.upper() == work_result
+        ), f"Workunit did not report the expected result:\n - payload: {payload}\n - work_result: {work_result}"
+
+        node1_controller.close()
+
+    def test_workunit_cmd_cancel(
+        self,
+        invoke_as_json,
+        default_receptor_controller_socket_file,
+        wait_for_work_finished,
+    ):
+        # Spawn a long running command
+        node1_controller = default_receptor_controller_socket_file
+        status = node1_controller.simple_command("status")
+
+        sleep_for = 9999  # in seconds
+
+        work = node1_controller.submit_work("sleep", str(sleep_for), node="node3")
+        state_result = work.pop("result")
+        state_unitid = work.pop("unitid")
+        assert state_result == "Job Started"
+
+        # Run and check cancel command
+        cancel_output = node1_controller.simple_command(f"work cancel {state_unitid}")
+        assert cancel_output["cancelled"] == state_unitid
+
+        # Get work list and check for the workunit detail state
+        work_list = node1_controller.simple_command(f"work list")
+        assert work_list[state_unitid]["Detail"] == "Killed"
```

Comment:
@fosterseth I think that I'm doing something wrong. I'm getting the following error:

```python
./tests/test_workunit.py::TestWorkUnit::test_workunit_cmd_cancel Failed: [undefined]AssertionError: assert 'Starting Worker' == 'Killed'
  - Killed
  + Starting Worker
```

And even if I sleep and wait for a state change, I get the following error:

```python
./tests/test_workunit.py::TestWorkUnit::test_workunit_cmd_cancel Failed: [undefined]AssertionError: assert 'Starting Worker' == 'Killed'
  - Killed
  + Launching command runner
```

Reply:
Adding the blocked tag until we investigate this error.